In [11]:
import pandas as pd

#Create a Data Frame
dataFrame = pd.read_csv('AggregateIncome.csv')

#Turn the strings in the income column into numerics
dataFrame['estimate__aggregate_household_income_in_the_past_12_months__'] = pd.to_numeric(dataFrame['estimate__aggregate_household_income_in_the_past_12_months__'], errors='coerce')

#Sort income decending from greatest to least
sortedIncome = dataFrame.sort_values(by='estimate__aggregate_household_income_in_the_past_12_months__', ascending=False)

# Display the top 10 neighborhoods in terms of aggregate income
print(sortedIncome[['estimate__aggregate_household_income_in_the_past_12_months__','Neighborhood']].head(10))


    estimate__aggregate_household_income_in_the_past_12_months__  \
15                                        659886300.0              
20                                        605635700.0              
23                                        577467500.0              
44                                        366945900.0              
50                                        314958500.0              
55                                        267100700.0              
58                                        249898900.0              
59                                        242405400.0              
60                                        241377900.0              
63                                        233098400.0              

           Neighborhood  
15  Squirrel Hill South  
20            Shadyside  
23  Squirrel Hill North  
44            Brookline  
50         Point Breeze  
55     Mount Washington  
58           Bloomfield  
59        Highland Park  
60           Gree

## Top 10 Neighborhoods in Total Income
1. Squirell Hill South - 659,886,300
2. Shadyside - 605,635,700
3. Squirrel Hill North - 577,467,500
4. Brookline - 366,945,900
5. Point Breeze - 314,958,500
6. Mount Washington - 267,100,700
7. Bloomfield - 249,898,900
8. Highland Park - 242,405,400
9. Greenfield - 241,377,900
10. South Side Flats - 233,098,400

In [7]:
#Create a Data Frame
dataFrame = pd.read_csv('Neighborhoods_with_SNAP_Data.csv')

#Remove the dollar sign and convert type to float
dataFrame['SNAP_All_csv_Median_Home_Value_'] = dataFrame['SNAP_All_csv_Median_Home_Value_'].replace(r'[\$,]', '', regex=True).astype(float)

#Sort income decending from greatest to least
sortedIncome = dataFrame.sort_values(by='SNAP_All_csv_Median_Home_Value_', ascending=False)

# Display the top 10 neighborhoods in terms of median home value
print(sortedIncome[['SNAP_All_csv_Median_Home_Value_','Neighborhood_2010_HOOD']].head(10))

    SNAP_All_csv_Median_Home_Value_ Neighborhood_2010_HOOD
65                         254667.0    Squirrel Hill North
1                          222423.0          North Oakland
4                          215900.0         Strip District
34                         164000.0           Point Breeze
55                         159700.0         Allegheny West
57                         153560.0              Shadyside
25                         134633.0    Squirrel Hill South
49                         115600.0             Friendship
68                         113000.0          Regent Square
37                          99900.0          Highland Park


## Top 10 Neighborhoods by Median Home Value
1. Squirrel Hill North - 254,667
2. North Oakland - 222,423
3. Strip District - 215,900
4. Point Breeze - 164,000
5. Allegheny West - 159,700
6. Shadyside - 153,560
7. Squirrel Hill South - 134,633
8. Friendship - 115,600
9. Regent Square - 113,000
10. Highland Park - 99,900

In [10]:
#Create a Data Frame
dataFrame = pd.read_csv('Neighborhoods_with_SNAP_Data.csv')

#Remove the percent sign and convert type to float
dataFrame['Edu__Attainment__Bachelor_s_Deg'] = dataFrame['Edu__Attainment__Bachelor_s_Deg'].replace(r'[\%,]', '', regex=True).astype(float)

#Sort education decending from greatest to least
sortedIncome = dataFrame.sort_values(by='Edu__Attainment__Bachelor_s_Deg', ascending=False)

# Display the top 10 neighborhoods in terms of education(bachelor's degree)
print(sortedIncome[['Edu__Attainment__Bachelor_s_Deg','Neighborhood_2010_HOOD']].head(10))

    Edu__Attainment__Bachelor_s_Deg Neighborhood_2010_HOOD
55                             44.8         Allegheny West
4                              33.4         Strip District
57                             32.3              Shadyside
49                             30.5             Friendship
70                             28.3     Point Breeze North
14                             27.6       South Side Flats
1                              27.4          North Oakland
25                             27.4    Squirrel Hill South
87                             26.7               West End
65                             26.7    Squirrel Hill North


## Top 10 Neighborhoods by Education (By percent with at least a bachelor's degree)
1. Allegheny West - 44.8
3. Strip District - 33.4
4. Shadyside - 32.3
5. Friendship - 30.5
6. Point Breeze North - 28.3
7. South Side Flats - 27.6
8. North Oakland - 27.4
9. Squirrel Hill South - 27.4
10. West End - 26.7
11. Squirrel Hill North - 26.7

## Conclusion
The same neighborhoods consistently show up on each list. Based on the data from these 3 sets, the new list will look like this:
1. Shadyside
2. Squirrel Hill North/Point Breeze 
4. Squirrel Hill South/Strip District
5. Allegheny West
6. North Oakland	 
8. Friendship
9. Brookline
10. South Side Flats
11. Mount Washington
12. Highland Park/Bloomfield
14. West End/Regent Square/Greenfield

In order to determine this conclusion, I assigned a value to each neighborhood based on its rank in each list. If it was not present on a certain list, it was automatically assigned an 11. Otherwise, it was assigned a point for its position on each list, from 1-10. Shadyside was the one that appeared highest on the list most often
| Neighborhood          | List 1 Rank | List 2 Rank | List 3 Rank | Total Rank |
|-----------------------|-------------|-------------|-------------|------------|
| Shadyside             | 3           | 6           | 2           | 11         |
| Squirrel Hill North   | 10          | 1           | 3           | 14         |
| Point Breeze          | 5           | 4           | 5           | 14         |
| Squirrel Hill South   | 8           | 7           | 1           | 16         |
| Strip District        | 2           | 3           | 11          | 16         |
| Allegheny West        | 1           | 5           | 11          | 17         |
| North Oakland         | 7           | 2           | 11          | 20         |
| Friendship            | 4           | 8           | 11          | 23         |
| Brookline             | 11          | 11          | 4           | 26         |
| South Side Flats      | 6           | 11          | 10          | 27         |
| Mount Washington      | 11          | 11          | 6           | 28         |
| Highland Park         | 11          | 10          | 8           | 29         |
| Bloomfield            | 11          | 11          | 7           | 29         |
| West End              | 9           | 11          | 11          | 31         |
| Regent Square         | 11          | 9           | 11          | 31         |
| Greenfield            | 11          | 11          | 9           | 31         |
